### ISSUE TITLE
TimeseriesGenerator for many-to-many architectures

### ISSUE THREAD

**Issue by kopytjuk** [keras-team/keras-preprocessing#287]
What do you think about introducing support for "many-to-many" architectures (see [here](http://karpathy.github.io/2015/05/21/rnn-effectiveness/)). In https://github.com/keras-team/keras/issues/11293 there is an issue about that functionality.

In the left side we have current implementation, on the right side the desired behaviour.

![image](https://user-images.githubusercontent.com/12762439/78949044-a77a1600-7aca-11ea-8c82-9d3e13eb4607.png)

**Comment by herzog-ch** [comment]
Yes please, this feature would help a lot in training RNNs.

**Comment by annaformaniuk** [comment]
Good idea - for me this would also be a helpful utility, replacing a lot of preprocessing code.

**Comment by nacheteam** [comment]
I have been looking how to do this in a generator fashion, any examples of how to do this with the actual functions? I suppose I need to extend the Sequence class but I'm not pretty sure how to do it... Thanks in advance!
